### ISSUE TITLE
Inbuilt comment feature not working properly for JSX files

### ISSUE THREAD

**Issue by Antony-AXS** [neovim/neovim#28830]
### Problem

when i type 'gcc; this is how it gets commented in a JSX file
![image](https://github.com/neovim/neovim/assets/99483158/308e9531-44b2-4588-a302-8c9d2580f73d)

### Steps to reproduce

![image](https://github.com/neovim/neovim/assets/99483158/45cd06f3-6c75-4555-a879-25176dc43470)

### Expected behavior

this is how it should be for JSX when commented.
![image](https://github.com/neovim/neovim/assets/99483158/96e60b1b-ab7d-4d87-b1b8-6ec7684666e6)

### Neovim version (nvim -v)

0.10

### Vim (not Nvim) behaves the same?

no

### Operating system/version

ubuntu 22.04

### Terminal name/version

 GNOME Terminal 3.44.0 using VTE 0.68.0 +BIDI +GNUTLS +ICU +SYSTEMD

### $TERM environment variable

xterm-256color

### Installation

appimage

**Comment by zeertzjq** [comment]
I don't think this is possible. The only way this can work is if there is a treesitter parser for javascriptreact, but there doesn't seem to be one.

**Comment by towry** [comment]
vi-mode on zed works well with `gcc` on tsx file.

**Comment by zeertzjq** [comment]
Oops, I was wrong. <https://github.com/tree-sitter/tree-sitter-javascript> does support .jsx files, but it seems that the comment string may change even though the filetype doesn't change. From https://github.com/numToStr/Comment.nvim/pull/133 it seems that this is quite a complicated problem.

**Comment by ribru17** [comment]
@zeertzjq Would you be opposed to a PR that sets the `{/* ... */}` comment string when the cursor is in a `(jsx_element)` node (maybe only checked in `(javascript|typescript)react` files)? Or would that be out of scope for the built-in comment module since the parser isn't bundled

**Comment by clason** [comment]
Yes, that would be out of scope. React files are such a mess, you _need_ a dedicated plugin to make sense of it. We shouldn't add one-off hacks for individual languages.

**Comment by folke** [comment]
This is a work-around to use `ts-context-commentstring` together with native comments:

```lua
      if vim.fn.has("nvim-0.10") == 1 then
        vim.schedule(function()
          local get_option = vim.filetype.get_option
          vim.filetype.get_option = function(filetype, option)
            return option == "commentstring" and require("ts_context_commentstring.internal").calculate_commentstring()
              or get_option(filetype, option)
          end
        end)
      end
```

An alternative would be to at least expose `vim._comment.get_commentstring` so that it can be easily overridden.

**Comment by folke** [comment]
Just created a tiny plugin to enhance the native comments with support for custom comment strings for certain treesitter node types.

It's an alternative to ts-context-commentstring.

https://github.com/folke/ts-comments.nvim
